src/Microsoft.AspNetCore.Mvc.Razor/DependencyInjection/MvcRazorMvcBuilderExtensions.cs

```diff
@@ -42,6 +45,28 @@ public static class MvcRazorMvcBuilderExtensions
         }
 
         /// <summary>
+        /// Registers tag helpers as services and changes the existing <see cref="ITagHelperActivator"/>
+        /// for an <see cref="ServiceBasedTagHelperActivator"/>.
+        /// </summary>
+        /// <param name="builder">The <see cref="IMvcBuilder"/> instance this method extends.</param>
+        /// <returns>The <see cref="IMvcBuilder"/> instance this method extends.</returns>
+        public static IMvcBuilder AddTagHelpersAsServices(this IMvcBuilder builder)
```

Comment:
This code is duplicated below for the corebuilder. Can we unify the logic and have both `AddTagHelperAsServices` call into a method that contains this? `DoSomethingUnified(PartManager, ServiceCollection)`